client/utils/auth.py

```diff
@@ -104,17 +109,19 @@ def update_storage(access_token, expires_in, refresh_token):
             'refresh_token': refresh_token
         }, fp)
 
-def authenticate(force=False):
-    """
-    Returns an oauth token that can be passed to the server for identification.
+def authenticate(args, force=False):
+    """Returns an OAuth token that can be passed to the server for
+    identification. If FORCE is False, it will attempt to use a cached token
+    or refresh the OAuth token. ARGS is the command-line arguments object.
     """
+    server = '{}://{}'.format('http' if args.insecure else 'https', args.server)
```

Comment:
The OAuth2 specification requires auth servers to use SSL - https://tools.ietf.org/html/rfc6749#section-3.1 

I don't think we should compromise on this - but I could see how insecure might help for local testing of the client. Further more, using insecure sends their `access_token` over the wire in plain-text anyway so this can't be much worse ...